bundles/org.openhab.binding.freeathomesystem/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,307 @@
+		<label>Read-only Heating Demand / Heating Value</label>
+		<description>Free@Home Thermostat Heating Demand Values (Read-only)</description>
+		<category>Temperature</category>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<!--Window Sensor Thing Type -->
+	<thing-type id="windowsensor" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Window Sensor</label>
+		<description>Window Sensor for Free@Home Binding</description>
+		<channels>
+			<channel id="window_state_channel" typeId="window_state_channel_type">
+				<label>Window State Channel</label>
+				<description>Window State / 1 - Open, 0 - Closed</description>
+			</channel>
+			<channel id="window_pos_channel" typeId="window_pos_channel_type">
+				<label>Window Position Channel</label>
+				<description>Position of Window</description>
+			</channel>
+		</channels>
+	</thing-type>
+
+	<!--Windod Sensor Channel Type -->
+	<channel-type id="window_state_channel_type">
+		<item-type>Number</item-type>
+		<label>Window State Channel</label>
+		<description>Free@Home Sensor Input Channel</description>
+		<category>Number</category>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="window_pos_channel_type">
+		<item-type>Number</item-type>
+		<label>Window Position Channel</label>
+		<description>Free@Home Sensor Input Channel</description>
+		<category>Number</category>
+		<state readOnly="true"></state>
+	</channel-type>
+
+
+	<!--Doorring Sensor Thing Type -->
+	<thing-type id="doorringsensor" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Door Ring Sensor</label>
+		<description>Door Ring for Free@Home Binding</description>
+		<channels>
+			<channel id="doorringsensor_channel" typeId="doorringsensor_channel_type">
+				<label>Ring State Channel</label>
+				<description>Ring State / 1 - Active, 0 - Note Active</description>
+			</channel>
+		</channels>
+	</thing-type>
+
+	<!--Doorring Channel Type -->
+	<channel-type id="doorringsensor_channel">
+		<item-type>Number</item-type>
+		<label>Ring Channel</label>
+		<description>Free@Home Ring Input Channel</description>
+		<category>Number</category>
+		<state readOnly="true"></state>
+	</channel-type>
+
+
+	<!-- Shutter Actuator Thing Type -->
+	<thing-type id="shutteractuator" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Shutter Actuator</label>
+		<description>Shutter Actuator for Free@Home Binding</description>
+		<channels>
+			<channel id="shutter_up_tigger_channel" typeId="shutter_trigger_channel_type">
+				<label>Shutter Actuator Up Trigger Channel</label>
+				<description>Shutter Actuator (UP) Trigger Channel for Free@Home Binding</description>
+			</channel>
+			<channel id="shutter_down_tigger_channel" typeId="shutter_trigger_channel_type">
```

Comment:
```suggestion
			<channel id="shutter_down_trigger_channel" typeId="shutter_trigger_channel_type">
```
Is this a typo?

Reply:
will be covered by the coding style changes

Reply:
Correction is done with coding style fixes